yo1k/tic_tac_toe/core/game.py

```diff
@@ -0,0 +1,296 @@
+from enum import Enum, auto
+from collections.abc import Sequence
+from typing import Optional
+from abc import ABC, abstractmethod
+
+
+class Mark(Enum):
+    X = auto()
+    O = auto()
+
+
+class Player:
+    def __init__(self, mark: Mark):
+        self.__mark = mark
+        self.__wins = 0
+
+    def mark(self) -> Mark:
+        return self.__mark
+
+    @property
+    def wins(self) -> int:
+        return self.__wins
+
+    @wins.setter
+    def wins(self, wins: int):
+        self.__wins = wins
+
+    def __str__(self):
+        return f"{type(self)}(" \
+               f"mark={self.__mark}," \
+               f"wins={self.__wins})"
```

Comment:
The formatting is broken here and in many other places in this file.

Reply:
Done.

Reply:
The formatting is still broken. For example, params of `__init__` methods are sometimes indented incorrectly. The assert statement on lines 128-130 is indented incorrectly. I do not specify all cases, because there are too many of them. I expect that you can find them and fix on your own.

Reply:
I don't make any changes until we discuss this. [Here](https://www.python.org/dev/peps/pep-0008/#:~:text=Code%20Lay%2Dout-,Indentation,-Use%204%20spaces) are the rules for indentation.

Reply:
Done.